apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx

```diff
@@ -47,6 +57,31 @@ export default class WorkshopEnrollmentSchoolInfo extends React.Component {
     this.props.onDelete(pendingDeleteId);
   }
 
+  handleScholarshipStatusChange(enrollment, selection) {
+    let pendingScholarshipUpdates = this.state.pendingScholarshipUpdates;
+    pendingScholarshipUpdates.push(enrollment.id);
+    this.setState({pendingScholarshipUpdates});
+    $.ajax({
+      method: 'POST',
+      url: `/api/v1/pd/enrollment/${
+        enrollment.id
+      }/scholarship_info?scholarship_status=${selection.value}`,
+      dataType: 'json'
+    }).done(data => {
+      let enrollments = this.state.enrollments;
+      const index = enrollments.findIndex(e => {
+        return e.id === data.id;
+      });
+      enrollments.splice(index, 1, data);
```

Comment:
This is another in-place modification of an array living in state.  Let's see if we can avoid this.